src/main/java/io/confluent/connect/hdfs/DataWriter.java

```diff
@@ -371,8 +384,13 @@ private void createDir(String dir) throws IOException {
   }
 
   @SuppressWarnings("unchecked")
-  private Format getFormat() throws ClassNotFoundException, IllegalAccessException, InstantiationException{
-    return  ((Class<Format>) Class.forName(connectorConfig.getString(HdfsSinkConnectorConfig.FORMAT_CLASS_CONFIG))).newInstance();
+  private Format getFormat() throws ClassNotFoundException, IllegalAccessException, InstantiationException,
+  NoSuchMethodException, SecurityException, IllegalArgumentException, InvocationTargetException{
+    final String className = connectorConfig.getString(HdfsSinkConnectorConfig.FORMAT_CLASS_CONFIG);
+    if (className.equals("io.confluent.connect.hdfs.avro.AvroFormat")) {
+        return (Format) Class.forName(className).getConstructor(HdfsSinkConnectorConfig.class).newInstance(new Object[] {connectorConfig});
```

Comment:
I think if we're going the route of needing configs, we should use an interface rather than detecting specific formats and using alternative constructors. Kafka has an `o.a.k.common.Configurable` that we could reuse.